### PR TITLE
Limit the length of Playlist folder

### PR DESCRIPTION
**Pull request by sangwoo108**
From the UI, the length is limited to 30 max characters.
With that, fixes text overflow in web ui

https://github.com/brave/brave-core/assets/5474642/94c8bb37-f83a-459e-9618-3aab569d1764

image

Resolves https://github.com/brave/brave-browser/issues/33618

## Submitter Checklist:

- [ ] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: browser/ui/views/playlist/playlist_action_dialogs.cc [hunk 1]
```diff
@@ -118,10 +118,66 @@ class TiledItemsView : public views::BoxLayoutView {
...
+
+  void Layout() override {
+    views::Textfield::Layout();
+
+    length_label_->SetBoundsRect(GetContentsBounds());
+  }
+
+ private:
+  void TruncateText() {
+    if (auto text = GetText(); text.length() > max_length_) {
```
Comment (fallaciousreasoning) [inline browser/ui/views/playlist/playlist_action_dialogs.cc:162]: Just a weird edge case, but what if you paste in some text which causes a Grapheme cluster to be cut in half? I guess it's probably okay to chop the grapheme in half?

**Review by fallaciousreasoning — APPROVED** [review]
LGTM, with one question about graphemes (which the text field might not even support) :)

**Review by diracdeltas — APPROVED** [review]

File: browser/ui/views/playlist/playlist_action_dialogs.cc (hunk 1 above)
Comment (sangwoo108) [reply browser/ui/views/playlist/playlist_action_dialogs.cc:162]: Wow, I didn't think about it. I thought the text.length should return the count of Grapheme, not the count of codes in the data, but it looks like the latter and could be chopped as you noted.
image

 But I believe it should be okay, also as you noted, as each segment of Grapheme cluster is valid code, like this one https://emojipedia.org/family-man-woman-girl#technical ?

Comment (sangwoo108) [reply browser/ui/views/playlist/playlist_action_dialogs.cc:162]: Ugh, I got this - 
image

This emoji consists of Person Bowing, Zero Width Joiner and ♂️ Male Sign but in the screen shot above, the code `U+1F647 | U+200D | U+2642 U+FE0F` seems to be truncated to `U+1F647 | U+200D | U+2642`. I guess if we want to solve this perfectly, we need use ICU module but it feels too much for now. Let's see what happens.

Comment (fallaciousreasoning) [reply browser/ui/views/playlist/playlist_action_dialogs.cc:162]: Yeah, I think it's probably so niche that it's not going to come up much :)

**Pull request merged**